1091. Shortest Path in Binary Matrix/solution.py

```diff
@@ -4,23 +4,77 @@ class Solution:
     # cells containing a 0. Each cell is connected to the 8 cells around it
     # that either share an edge or a corner. If no path is found, return -1.
     def shortestPathBinaryMatrix(self, grid: List[List[int]]) -> int:
+        # Check that top left and bottom right cells contain 0s. Otherwise no clear
+        # path can exist, so return -1
+        gridHeight = len(grid)
+        gridWidth = len(grid[0])
+        
+        if grid[0][0] != 0 or grid[gridHeight-1][gridWidth-1] != 0:
```

Comment:
grid[gridHeight-1][gridWidth-1] can be converted into grid[-1][-1].  Also, reading the constraints again; it has to be 1 in order for it to not be 0.

Reply:
ooo the -1 trick is a good idea. I'll use that. What has to be 1 btw, the min length and width of the grid?

Reply:
Since the matrix is a binary matrix, each cell entry has to be 1 or 0.

Reply:
Oh... I see. I don't think it's worth changing. Thanks tho.